### DISCUSSION TITLE
Horizontal Scrollbar In Repeater

### DISCUSSION THREAD

**Discussion by danderson2692** [Budibase/budibase#6754] [Help] [answered]
I'm creating an app to act as a kiosk for dogs who are available for adoption, allowing adopters to view more info about them and to allow staff/volunteers to log info like medical/behavioral info.

Is it possible to force a repeater to use a horizontal scrollbar rather than a vertical one? I'd like to be able to make these containers close to the full screen height so there is enough space for 2 or 3 dogs to appear on the screen at once, but I don't want to use pagination to flip through the list of dogs, and I think navigation will be smoother with horizontal scrolling.

This is how the landing page looks:
![image](https://user-images.githubusercontent.com/16950976/179360794-4aac3998-b63d-4e76-8f66-8e969ca4f26b.png)

Thanks in advance!

**Comment by poirazis** [comment]
You can set the Custom CSS of the Repeater to allow for horizontal overflow
More info here : https://www.w3schools.com/cssref/css3_pr_overflow-x.asp

**Reply by danderson2692** [reply]
Hi poirazis,

I've been playing with this for a while but haven't been able to figure out how to make it work. I've found that I have to set the overflow of the screen (overflow: scroll hidden), then in the repeater I have to set both the position and the overflow settings in order to be able to change it, but I haven't been able to figure out how to get the third record to appear on the same row after setting overflow: scroll hidden. When I do this, I can't scroll vertically, but I also can't scroll horizontally and I can still see the 3rd record beneath the first two. I also have to set position: absolute or fixed which move the repeater into the wrong position on the screen; setting position: static causes the vertical scrollbar not to be hidden.

Do you have any suggestions of how I can correct this or examples showing the same thing?

**Reply by poirazis** [reply]
I think you should put a Container inside the Repeater, and put you data inside the Container.
Then put the Custom CSS of this container to `overflow-x: scroll;`
This would give a horizontal scrollbar to the Container and would not affect the Repeater.

**Reply by danderson2692** [reply]
I've tried to do that, and to apply it at the Data Provider level that the repeater is inside of, but it still shows the same behavior. The overall hierarchy I have for the screen is as follows with all of them set with custom CSS, overflow: scroll hidden; (I'm using this instead of separate overflow-x and -y because this did allow me to hide the vertical scrollbar with one line):

- Screen
- Data Provider
- Repeater
- Container
- Images, Paragraphs, Buttons (without custom CSS)

Since I don't have the position set in the CSS at any level, the vertical scrollbar is still visible and the horizontal one still isn't.

This is the thread that I found which led me to try setting the position: https://stackoverflow.com/questions/61302178/how-does-spectrum-chat-hide-scrollbar-for-secondary-channel-section

**Reply by poirazis** [reply]
If I get it right, you want to have a slider behaviour?

**Reply by danderson2692** [reply]
Sort of - I'm hoping to be able to scroll through the records horizontally, sort of like in the Data view when you have too many columns to show at once.
![image](https://user-images.githubusercontent.com/16950976/179420338-6005dfbb-8380-4646-91bb-cec3676161c2.png)

**Comment by poirazis** [comment, marked as answer]
[TestApp-export-1658132528474.txt](https://github.com/Budibase/budibase/files/9130462/TestApp-export-1658132528474.txt)
Please inspect the Custom CSS of all the Elements in this TestApp.
I have forced paragraphs not to wrap and the container to scroll horizontally.

The Custom CSS of the Repeater is 
`flex-wrap : nowrap;
overflow-x : scroll;`

<img width="1191" alt="Screenshot 2022-07-18 at 11 22 38 AM" src="https://user-images.githubusercontent.com/80512170/179472192-00ce7d4c-b3bc-4fc1-98bb-66a1822d0328.png">
[TestApp-export-1658132528474.txt](https://github.com/Budibase/budibase/files/9130457/TestApp-export-1658132528474.txt)

**Reply by melohagan** [reply]
@danderson2692 This answer looks like it'll hopefully work for you! 

As a side note, you can create an image carousel if that is something you're interested in: https://github.com/Budibase/budibase/discussions/4985#discussioncomment-2444397

**Reply by danderson2692** [reply]
@poirazis thanks for all of your help and your patience with me! One of the main pieces that I was missing the flex-wrap: nowrap; which got me very close. Ultimately I ended up needing the following:
Repeater:
flex-wrap: nowrap;
overflow: scroll hidden;

Container:
flex-shrink: 0;

I need to play with the formatting to make it look prettier, but this is the result:
![image](https://user-images.githubusercontent.com/16950976/179646249-0f582067-b2eb-4ee6-9343-fdbfd1e4d98b.png)

**Reply by wb-ran** [reply]
Hi guys! Is this still possible? Or did the behaviour of the repeater block change?
I tried to replicate it, it seems to me that the repeater always wraps on it's own, no matter what the CSS is.. (?)

CSS of Repeater:
flex-wrap : nowrap; 
overflow-x : scroll;

![image](https://github.com/user-attachments/assets/f70a0b1c-076f-4be8-b21b-def88900ae53)

**Reply by melohagan** [reply]
Hey @wb-ran You may need to [eject](https://docs.budibase.com/docs/blocks#ejecting-blocks) the Repeater Block and then apply the styling to the Repeater component.

**Reply by wb-ran** [reply]
Hi @melohagan, you're absolutely correct, that was the problem! Thank you!
